### ISSUE TITLE
Entropy crop behaves differently between JPG and PNG files with alpha transparency

### ISSUE THREAD

**Issue by ChadLevy** [JimBobSquarePants/ImageProcessor#657]
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have verified that I am running the latest version of ImageProcessor
- [X] I have verified if the problem exist in both `DEBUG` and `RELEASE` mode
- [X] I have searched [open](https://github.com/JimBobSquarePants/ImageProcessor/issues) and [closed](https://github.com/JimBobSquarePants/ImageProcessor/issues?q=is%3Aissue+is%3Aclosed) issues to ensure it has not already been reported

### Description
I have a scenario where I'm removing a watermark from images (my license enables me to do this) on the fly using ImageProcessor.Web. To accomplish this, I crop the image by 5%, then run EntropyCrop with a value of 128, then resize it. The query in its entirety:

    ?crop=5,5,5,5&cropmode=percentage&entropycrop=128&width=1100&height=1100&mode=max

When applying this query to JPG images, it works as expected:

Original:

![image](https://user-images.githubusercontent.com/909190/37572132-cf0112b8-2ac3-11e8-8361-0d289ed64687.png)

Processed:

![image](https://user-images.githubusercontent.com/909190/37572146-20bb5cd0-2ac4-11e8-858e-736932486b1e.png)

With a PNG, however, the entry crop appears to be too aggressive:

Original:

![image](https://user-images.githubusercontent.com/909190/37572178-8dad1ebe-2ac4-11e8-8316-7b6d327c4d3e.png)

Processed:

![image](https://user-images.githubusercontent.com/909190/37572174-7e53367e-2ac4-11e8-8779-cef75ed2b30a.png)

As you can see, the tire at the bottom has been cut off.

I've tried various values for `entropycrop` to no avail. It seems to have no effect on a PNG.

### Steps to Reproduce

1. Clone the example repository:

      `git clone https://github.com/ChadLevy/imageprocessor-png-issue.git`

2. Open WebApplication1.sln in Visual Studio. Restore nuget packages if necessary. Run the web application without debugging.

3. Navigate to http://localhost:53248/content/sample-images/sample.jpg?crop=5,5,5,5&cropmode=percentage&entropycrop=500&width=1100&height=1100&mode=max. Observe that the image has been cropped correctly.

4. Navigate to http://localhost:53248/content/sample-images/sample.png?crop=5,5,5,5&cropmode=percentage&entropycrop=500&width=1100&height=1100&mode=max. Observe the image has not been cropped correctly (the bottom of the tire has been cut off).

### System Configuration

- ImageProcessor version: 2.6.0
- ImageProcessor.Web version (if applicable): 4.9.1.2
- Other ImageProcessor packages and versions:
  - ImageProcessor.Web.Config 2.4.0
  - ImageProcessor.Web.Plugins.AzureBlobCache 1.4.1.2
  - ImageProcessor.Web.PostProcessor 1.3.0

- Environment (Operating system, version and so on): 
  - Windows 10 Pro, x64, version 1709 (build 16299.309)
  - Visual Studio 2017 15.5.6
- .NET Framework version: 4.7.02557
- Additional information:
  - _Please see the config files in the included sample repository._

**Comment by JimBobSquarePants** [comment]
Hi @ChadLevy 

Thanks for all the detail, I appreciate it. 👍 

I've not dug into this at all as the process should be file format independent but something jumped out at me straight away.

`http://localhost:53248/content/sample-images/sample.jpg?crop=5,5,5,5&cropmode=percentage&entropycrop=500&width=1100&height=1100&mode=max.`

You're passing far too high a value **500** to the entropy cropper. The value is a threshold and should be between **0-255** since that it the byte value range of a pixel component withing the `Color` struct.

Pre-cropping is redundant also unless you are chopping off some sort of border first?

**Comment by ChadLevy** [comment]
I was trying a bunch of random values for entropycrop and mistakenly pasted
one of those. The value I’ve been using is 128. Curiously adjusting that
value seems to have no effect.

Regarding the double crop, the first one removes a watermark in the
lower-right which then enables the entropycrop to crop away the unnecessary
white space.

**Comment by JimBobSquarePants** [comment]
Thanks for the update,

I'll try and have a look later on for you. The entropy cropper is file format independant as it just looks at the pixel data.

**Comment by ChadLevy** [comment]
Well, I got it to work (mostly) with an `entropycrop` value of 1. The shadow at the bottom tends to get cut off a little bit but I can live with that. An `entropycrop` value of 1 for a PNG with alpha transparency seems to equate to 128 for a JPG in my sample set.

In my testing I never considered using 1 because using 0 seemed to have no effect. My hopes in writing a detailed description was to catch such a derpy mistake. Guess not!

**Comment by JimBobSquarePants** [comment]
I'm gonna reopen this as I want to do some investigation as the numbers don't fit right to me. I'll let you know how I get on.

**Comment by JimBobSquarePants** [comment]
@ChadLevy 

I've been running pretty extensive testing using the first uploaded png and it's working perfectly.
![original](https://user-images.githubusercontent.com/909190/37572132-cf0112b8-2ac3-11e8-8361-0d289ed64687.png)

The second uploaded original png doesn't contain any transparency an the checkered pattern is part of the image so I can't test the cropper with it. Do you have the proper png that you used to generate the over-cropped output?

![kia](https://user-images.githubusercontent.com/385879/38143058-9a576622-348a-11e8-9667-18301994cd9f.JPG)

**Comment by ChadLevy** [comment]
Yeah, the png included in the sample repo has alpha transparency:

https://github.com/ChadLevy/imageprocessor-png-issue/blob/master/WebApplication1/WebApplication1/Content/sample-images/sample.png

**Comment by ChadLevy** [comment]
Thanks for looking into this @JimBobSquarePants. Your explanation helps a lot. I suspected it was something along these lines. I appreciate your help!

**Comment by JimBobSquarePants** [comment]
No worries, happy to 😄
